src/model/channel/message.rs

```diff
@@ -195,7 +195,7 @@ impl Message {
 
         // First replace all user mentions.
         for u in &self.mentions {
-            result = result.replace(&u.mention(), &u.distinct());
+            result = result.replace(&u.mention(), &format!("@{}", u.distinct()));
```

Comment:
If we're going to change this, let's make `format!("@{}", u.distinct())` faster:

```rust
let mut at_distinct = String::with_capacity(38); // 1 + max length of username (32) + 1 + length of discrim (4)
at_distinct.push('@');
at_distinct.push_str(&u.name);
at_distinct.push('#');
at_distinct.push_str(&u.discriminator);
```

compared to the committed code, this is nearly 90% faster:
```
test tests::one ... bench:          20 ns/iter (+/- 0)
test tests::two ... bench:         180 ns/iter (+/- 2)
```